falcon_multiqc/commands/query.py

```diff
@@ -68,50 +61,87 @@ def query_select(session, columns, tool_metric_filters, multiqc):
                     c = func.max(RawData.metrics[tm[1]].astext).label(tm[1])
                     c.quote = True
                     select_cols.append(c)
+            else:
+                select_cols.append(RawData.qc_tool)
     if multiqc:
         # Need to add batch paths.
         select_cols.append(Batch.path)
 
     query = Query(select_cols, session=session)
-
-    # Add the table joins needed for the given column selection.
-    if 'sample' in columns or 'batch' in columns:
-        # We need Batch for batch.path for both sample and batch.
-        query = query.join(Batch, Sample.batch_id == Batch.id)
-    
-    if 'tool-metric' in columns: 
-        query = query.join(RawData, Sample.id == RawData.sample_id)
-
-    if 'cohort' in columns:
-        query = query.join(Cohort, Sample.cohort_id == Cohort.id)
+    join['joined'].add(columns[0]) # the first select of a given query doesn't need to be explicitly joined
+
+    ### ================================= JOIN  ==========================================####
+    # Add the table joins needed for the given column selection or filtering.
+    # TODO condense this with a function potentially 
+
+    if 'sample' in join['joins'] and 'sample' not in join['joined']:
+        if 'tool-metric' in join['joined']:
+            query = query.join(Sample, Sample.id == RawData.sample_id)
+        elif 'batch' in join['joined']:
+            query = query.join(Sample, Batch.id == Sample.batch_id)
+        elif 'cohort' in join['joined']:
+            query = query.join(Sample, Sample.cohort_id == Cohort.id)
+        join['joined'].add('sample')
+
+    # for multiqc we need Batch for batch.path.
```

Comment:
Tiny nit: can you make sure comments are capitalized and have fullstop everywhere, see line 290 too.